src/main/java/com/fasterxml/jackson/core/io/SegmentedStringWriter.java

```diff
@@ -37,34 +36,84 @@ public Writer append(char c) {
     @Override
     public Writer append(CharSequence csq) {
         String str = csq.toString();
-        _buffer.append(str, 0, str.length());
```

Comment:
Since basic `Writer` exposes `IOException`, is there a reason we can't do that here? Does caller not (yet) expose them?

Reply:
Quick look suggests that we can just throw `IOException` since it's `ObjectMapper` and `ObjectWriter` that use it but only like regular `Writer` (at least during write path).

Reply:
changed